src/globus_sdk/scopes/representation.py

```diff
@@ -89,8 +94,16 @@ def with_dependencies(self, other_scopes: t.Iterable[Scope]) -> Scope:
 
         :param other_scopes: The scopes upon which the current scope depends.
         """
+        other_scopes_tuple = tuple(other_scopes)
+        for i, item in enumerate(other_scopes_tuple):
+            if not isinstance(item, Scope):
+                raise TypeError(
+                    "Scope.with_dependencies() takes "
+                    "an Iterable of Scopes as its input. "
```

Comment:
I don't think that Python class capitalization is needed here. Suggesting:

```suggestion
                    "an iterable of Scopes as its input. "
```